packages/dev/core/src/Materials/greasedLineSimpleMaterial.ts

```diff
@@ -0,0 +1,511 @@
+            width: this._width,
+        };
+
+        this._colors && (greasedLineMaterialOptions.colors = this._colors);
+
+        serializationObject.greasedLineMaterialOptions = greasedLineMaterialOptions;
+
+        return serializationObject;
+    }
+
+    /**
+     * Parses a serialized objects
+     * @param source serialized object
+     * @param scene scene
+     * @param _rootUrl root url for textures
+     */
+    public parse(source: any, scene: Scene, _rootUrl: string): void {
+        // TODO: super.parse?
+        const greasedLineMaterialOptions = <GreasedLineMaterialOptions>source.greasedLineMaterialOptions;
+
+        this._colorsTexture?.dispose();
+
+        if (greasedLineMaterialOptions.colors) {
+            this._createColorsTexture(`${this.name}-colors-texture`, greasedLineMaterialOptions.colors);
+        } else {
+            GreasedLineSimpleMaterial._PrepareEmptyColorsTexture(scene);
+        }
+
+        greasedLineMaterialOptions.color && (this.color = greasedLineMaterialOptions.color);
+        greasedLineMaterialOptions.colorDistributionType && (this.colorsDistributionType = greasedLineMaterialOptions.colorDistributionType);
+        greasedLineMaterialOptions.colorsSampling && (this.colorsSampling = greasedLineMaterialOptions.colorsSampling);
+        greasedLineMaterialOptions.colorMode && (this.colorMode = greasedLineMaterialOptions.colorMode);
+        greasedLineMaterialOptions.useColors && (this.useColors = greasedLineMaterialOptions.useColors);
+        greasedLineMaterialOptions.visibility && (this.visibility = greasedLineMaterialOptions.visibility);
+        greasedLineMaterialOptions.useDash && (this.useDash = greasedLineMaterialOptions.useDash);
+        greasedLineMaterialOptions.dashCount && (this.dashCount = greasedLineMaterialOptions.dashCount);
+        greasedLineMaterialOptions.dashRatio && (this.dashRatio = greasedLineMaterialOptions.dashRatio);
+        greasedLineMaterialOptions.dashOffset && (this.dashOffset = greasedLineMaterialOptions.dashOffset);
+        greasedLineMaterialOptions.width && (this.width = greasedLineMaterialOptions.width);
+        greasedLineMaterialOptions.sizeAttenuation && (this.sizeAttenuation = greasedLineMaterialOptions.sizeAttenuation);
+        greasedLineMaterialOptions.resolution && (this.resolution = greasedLineMaterialOptions.resolution);
+    }
+
+    /**
+     * A minimum size texture for the colors sampler2D when there is no colors texture defined yet.
+     * For fast switching using the useColors property without the need to use defines.
+     * @param scene Scene
+     */
+    private static _PrepareEmptyColorsTexture(scene: Scene) {
+        if (!this._EmptyColorsTexture) {
+            const colorsArray = new Uint8Array(4);
+            this._EmptyColorsTexture = new RawTexture(colorsArray, 1, 1, Engine.TEXTUREFORMAT_RGBA, scene, false, false, RawTexture.NEAREST_NEAREST);
+            this._EmptyColorsTexture.name = "grlEmptyColorsTexture";

```

Comment:
nit - I find `this` in a static function misleading. Though it is valid, i know. I think changing it to the class name will make it clearer to read.

Reply:
How could I... I never do this... :-) When reading the texture I use the classname to access the texture.